pandas/core/dtypes/cast.py

```diff
@@ -1654,12 +1654,33 @@ def cast_scalar_to_array(
 
     """
     if dtype is None:
-        dtype, fill_value = infer_dtype_from_scalar(value)
+        dtype, value = infer_dtype_from_scalar(value)
     else:
-        fill_value = value
+        if not isinstance(dtype, np.dtype):
+            dtype = np.dtype(dtype)
+        empty = shape and not any(shape)
+        # rem: type coercion if empty: sometimes yes, sometimes no ?
```

Comment:
what is "rem"?

Reply:
oh, just remark

Reply:
but yes, sorry, this is a real question:  in the case of empty DF/Series init, is this non-uniformity in dtype casting wanted ?

Reply:
use TODO (or don't use rem its not common nomeclature)